corehq/apps/hqwebapp/static/hqwebapp/js/ui_elements/ui-element-input-map.js

```diff
@@ -81,12 +81,11 @@ hqDefine('hqwebapp/js/ui_elements/ui-element-input-map', [
                     });
 
                 }
-                let left_side = map_key ?
-                    '<strong>' + $('<div>').text(map_key).html() + '</strong>' :
-                    '<i>' + django.gettext('blank') + '</i>';
-                this.$noedit_view.html(left_side + ' &rarr; ' + (
-                    map_val ? $('<div>').text(map_val).html() : django.gettext('(empty)')
-                ));
+                if (map_key && !map_key.trim()) map_key = `"<span style="white-space: pre;">${map_key}</span>"`;
```

Comment:
Ah, oof. If I use `DOMPurify.sanitize` that strips the offending content vs what had been there escapes it. I like the clarity of `DOMPurify` (since obviously the point of the successive `.text` `.html` jquery calls passed me by) but I can't in a brief look find a way to make `DOMPurify` simply escape rather than strip...is it really OK to strip out offending content or is there way I'm missing to make `DOMPurify` escape?

Reply:
I think it's fine to strip any tags - I can't imagine anyone was relying on the behavior of showing the escaped HTML.

Reply:
OK I pushed up a change that uses `DOMPurify`. I did it in a way that fully-stripped values get turned into the italic blank, not sure if that is best or not. (The alternative is fully-stripped values just appear not to be there leaving possibly a lone arrow. Neither seems particularly great but perhaps this is an extreme edge case?)

Reply:
I think it's an edgy enough edge case not to worry about.